evil-mc.el

```diff
@@ -43,18 +43,35 @@
 (require 'evil-mc-region)
 
 (defcustom evil-mc-mode-line
-  `(:eval (cond ((<= (evil-mc-get-cursor-count) 1)
-                 (format ,(propertize " %s") evil-mc-mode-line-prefix))
-                ((evil-mc-frozen-p)
-                 (format ,(propertize " %s(paused):%d" 'face '(:inherit cursor :inverse-video t))
-                         evil-mc-mode-line-prefix (evil-mc-get-cursor-count)))
-                (t
-                 (format ,(propertize " %s:%d" 'face 'cursor)
-                         evil-mc-mode-line-prefix (evil-mc-get-cursor-count)))))
-  "Cursors indicator in the mode line."
-  :group 'evil-mc)
-
-(put 'evil-mc-mode-line 'risky-local-variable t)
+  `(:eval
+    (let ((mode-line-text-prefix (concat " " evil-mc-mode-line-prefix)))
+      (cond ((<= (evil-mc-get-cursor-count) 1)
+             (when evil-mc-one-cursor-show-mode-line-text
+               mode-line-text-prefix))
+            (t  ; two or more cursors
+             ;; mode line text
+             (let ((mode-line-text
```

Comment:
Yeah, anything that improves readability is great. Do you have a suggestion for how the function could be written?